### ISSUE TITLE
Re-Indexing Fails on Chef Infra Upgrade to 14.1 from 13.2

### ISSUE THREAD

**Issue by tas50** [chef/chef-server#2389]
This is a customer bug reported at:
https://github.com/chef/customer-bugs/issues/339

More:
https://getchef.zendesk.com/agent/tickets/27388

```
We are upgrading Chef Infra Server from 13.2.0 to 14.1.0. The upgrade completed and appears to have worked. A ‘chef-server-ctl version’ command reports the 14.1.0 version. Several other ‘chef-server-ctl’ commands work as well. A ‘chef-server-ctl test’ command also completed successfully:

. . .
Pending: (Failures listed here are expected and do not affect your suite's status)

1) opscode-account user association user not in org can be invited to the org by an admin when the inviting admin is removed from the org, invites issued by that admin cannot be accepted
 # Known failure: passes w/ 200 b/c no USAG cleanup performed for deleted user
 # ./spec/api/account/account_association_spec.rb:662

2) opscode-account user association user not in org can be invited to the org by an admin when the inviting admin is removed from the system, invites issued by that admin can't by accepted
 # Known failure: passes w/ 200 b/c no USAG or other group cleanup performed for deleted user
 # ./spec/api/account/account_association_spec.rb:671

Finished in 2 minutes 2.9 seconds (files took 11.21 seconds to load)
172 examples, 0 failures, 2 pending

However, the re-indexing failed.

Here is a portion of the output from the ‘chef-server-ctl upgrade’ command:

. . .
[private-chef-upgrade] - Sleeping for 15 seconds while services comes online...
[private-chef-upgrade] - All orgs are in the 503 mode...
[private-chef-upgrade] - Migrating indexed search data...
[private-chef-upgrade] - Running Command: chef-server-ctl reindex -a -d -t with options {}
Reindexing all organizations
- Disabling the Chef API.
- Reindexing.
escript: exception error: no match of right hand side value
 {error,{conn_failed,{error,econnrefused}}}
Failed to reindex data for abc!
- Re-enabling the Chef API
0.471267848 seconds to reindex.
[private-chef-upgrade] - Stopping Services ["elasticsearch", "nginx", "oc_bifrost", "opscode-erchef", "redis_lb"]
ok: down: elasticsearch: 0s, normally up
down: elasticsearch: 2s, normally up; run: log: (pid 17346) 198s
down: elasticsearch: 4s, normally up; run: log: (pid 17346) 200s
ok: down: nginx: 0s, normally up
. . .
[private-chef-upgrade] - Sleeping for 10 seconds while services stop...
[private-chef-upgrade] - Finished Migration 1.35 in 149.61 seconds
Chef Server Upgraded!

Running the ‘reindex’ command outside of the upgrade produces the same result:


# chef-server-ctl reindex -d -t abc
Reindexing orgs: ["abc"]
- Disabling the Chef API.
- Reindexing.
escript: exception error: no match of right hand side value
 {error,{conn_failed,{error,econnrefused}}}
Failed to reindex data for abc!
- Re-enabling the Chef API
0.33425655 seconds to reindex.
```

Aha! Link: https://chef.aha.io/features/SH-3059

**Comment by lbakerchef** [comment]
Prajakta Purohit  12:43 AM
@hh - I looked a bit into the error and the logs they sent. Some questions/observations.
It looks like they are running a standalone setup Chef Infra Server and using internal elasticsearch and postgresql.
The error they are seeing in reindex seems to be because https://github.com/chef/chef-server/blob/master/src/oc_erchef/priv/reindex-opc-organization#L142 returns an error and does not pattern match as expected. (I will create a ticket to bee able to throw a more meaningful error in case of a failure here).
Question for them : is epmd running? Is the port for epmd open? I would expect to see something like
root@api:~# ps ax | grep epmd
 1614 ?        S      0:00 /opt/opscode/embedded/lib/erlang/erts-10.6/bin/epmd -daemon
 2341 pts/0    S+     0:00 grep --color=auto epmd
root@api:~# /opt/opscode/embedded/lib/erlang/erts-10.6/bin/epmd -names
epmd: up and running on port 4369 with data:
name erchef at port 36661
name bookshelf at port 43592
name oc_bifrost at port 36802
if all is ok on that front.
Updating from 13 -> 14 implies they are moving away from rabbitmq. It would be good to confirm none of the rabbitmq processes are stuck / blocking epmd.
is localhost resolvable to 127.0.0.1 on their system? (Although in ourcase we use 127.0.0.1 so this should not be an issue.)
Is their machine on ipv4 or ipv6?
Answers to ^^ might help us solving the issue further. (edited) 
src/oc_erchef/priv/reindex-opc-organization:142
    {ok, _} = net_kernel:start([?SELF, longnames]),
<https://github.com/chef/chef-server|chef/chef-server>chef/chef-server | Added by GitHub
12:45
the issue information is available at: https://getchef.zendesk.com/agent/tickets/27388
logfile from the ticket here for ref:
```
Script ri.sh:
echo; echo "************* Status *************"
echo; echo "chef-server-ctl status"; chef-server-ctl status
#
echo; ls -l /var/log/opscode/opscode-erchef/current
echo; ls -l /var/log/opscode/elasticsearch/current
#
echo; echo "************* Reindex *************"
echo; date; echo; echo "chef-server-ctl reindex -a -d -t"; chef-server-ctl reindex -a -d -t
echo;
#
sleep 10
echo; echo; echo "*************  Identify all changed files under /var and /opt in the last minute   *************"
echo; date; echo "find /var -cmin -1 -ls"; find /var -cmin -1 -ls
echo; date; echo "find /opt -cmin -1 -ls"; find /opt -cmin -1 -ls; echo
#
echo; echo "************* Post-Reindex Log Dump *************"
#
echo; ls -l /var/log/opscode/opscode-erchef/current
echo; ls -l /var/log/opscode/elasticsearch/current
echo; ls -l /var/log/opscode/oc_bifrost/requests.log.1
echo; ls -l /var/log/opscode/opscode-erchef/requests.log.1
echo; ls -l /var/log/opscode/redis_lb/current
echo; ls -l /var/log/opscode/nginx/access.log
#
echo; echo "************* Post-Reindex Log Dump *************"
echo; echo tail -10 /var/log/opscode/opscode-erchef/current; tail -10 /var/log/opscode/opscode-erchef/current
echo; echo tail -3 /var/log/opscode/opscode-erchef/requests.log.1; tail -3 /var/log/opscode/opscode-erchef/requests.log.1
echo; echo tail -10 /var/log/opscode/elasticsearch/current; tail -10 /var/log/opscode/elasticsearch/current
echo; echo tail -10 /var/log/opscode/oc_bifrost/requests.log.1; tail -10 /var/log/opscode/oc_bifrost/requests.log.1
echo; echo tail -10 /var/log/opscode/nginx/access.log; tail -10 /var/log/opscode/nginx/access.log
echo; echo tail -10 /var/log/opscode/redis_lb/current; tail -10 /var/log/opscode/redis_lb/current
#
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[root@myhost0010 ~]# ./ri.sh
************* Status *************
chef-server-ctl status
run: bookshelf: (pid 31557) 369898s; run: log: (pid 1699) 681023s
run: elasticsearch: (pid 29118) 171609s; run: log: (pid 1720) 681023s
run: nginx: (pid 30177) 170835s; run: log: (pid 1712) 681023s
run: oc_bifrost: (pid 31655) 369897s; run: log: (pid 1698) 681023s
run: oc_id: (pid 31683) 369896s; run: log: (pid 1711) 681023s
run: opscode-erchef: (pid 31796) 369896s; run: log: (pid 1716) 681023s
run: postgresql: (pid 31878) 369895s; run: log: (pid 1700) 681023s
run: redis_lb: (pid 29097) 171610s; run: log: (pid 1708) 681023s
-rw-r--r--. 1 opscode opscode 514858 Mar 22 08:30 /var/log/opscode/opscode-erchef/current
-rw-r--r--. 1 opscode opscode 78408 Mar 24 17:05 /var/log/opscode/elasticsearch/current
************* Reindex *************
Fri Mar 26 15:15:04 EDT 2021
chef-server-ctl reindex -a -d -t
Reindexing all organizations
- Disabling the Chef API.
- Reindexing.
escript: exception error: no match of right hand side value 
                 {error,{conn_failed,{error,econnrefused}}}
Failed to reindex data for abc!
- Re-enabling the Chef API
0.36082007 seconds to reindex.
*************  Identify all changed files under /var and /opt in the last minute   *************
Fri Mar 26 15:15:16 EDT 2021
find /var -cmin -1 -ls
201426659 34980 -rw-r--r--   1 opscode  opscode  35810671 Mar 26 15:15 /var/log/opscode/oc_bifrost/requests.log.1
163866 42380 -rw-r--r--   1 opscode  opscode  43390880 Mar 26 15:15 /var/log/opscode/opscode-erchef/requests.log.1
201426983 60224 -rw-r--r--   1 opscode  opscode  61612890 Mar 26 15:15 /var/log/opscode/nginx/access.log
   499    0 drwx------   2 opscode-pgsql root          164 Mar 26 15:15 /var/opt/opscode/postgresql/9.6/data/pg_stat_tmp
163261    4 -rw-------   1 opscode-pgsql opscode-pgsql     3385 Mar 26 15:15 /var/opt/opscode/postgresql/9.6/data/pg_stat_tmp/db_0.stat
163258    4 -rw-------   1 opscode-pgsql opscode-pgsql     1532 Mar 26 15:15 /var/opt/opscode/postgresql/9.6/data/pg_stat_tmp/global.stat
163260   48 -rw-------   1 opscode-pgsql opscode-pgsql    47325 Mar 26 15:15 /var/opt/opscode/postgresql/9.6/data/pg_stat_tmp/db_16387.stat
163262   24 -rw-------   1 opscode-pgsql opscode-pgsql    21975 Mar 26 15:14 /var/opt/opscode/postgresql/9.6/data/pg_stat_tmp/db_17441.stat
163265   32 -rw-------   1 opscode-pgsql opscode-pgsql    31777 Mar 26 15:14 /var/opt/opscode/postgresql/9.6/data/pg_stat_tmp/db_17092.stat
163263   20 -rw-------   1 opscode-pgsql opscode-pgsql    18933 Mar 26 15:14 /var/opt/opscode/postgresql/9.6/data/pg_stat_tmp/db_1.stat
163266   20 -rw-------   1 opscode-pgsql opscode-pgsql    18933 Mar 26 15:14 /var/opt/opscode/postgresql/9.6/data/pg_stat_tmp/db_16384.stat
163259   20 -rw-------   1 opscode-pgsql opscode-pgsql    18933 Mar 26 15:14 /var/opt/opscode/postgresql/9.6/data/pg_stat_tmp/db_13325.stat
    96   52 drwxrwxrwt  17 root     root        28672 Mar 26 15:15 /var/tmp
  7438   32 -rw-------   1 opscode  opscode     32768 Mar 26 15:15 /var/tmp/hsperfdata_opscode/29118
Fri Mar 26 15:15:16 EDT 2021
find /opt -cmin -1 -ls
************* Post-Reindex Log Dump *************
-rw-r--r--. 1 opscode opscode 514858 Mar 22 08:30 /var/log/opscode/opscode-erchef/current
-rw-r--r--. 1 opscode opscode 78408 Mar 24 17:05 /var/log/opscode/elasticsearch/current
-rw-r--r--. 1 opscode opscode 35810671 Mar 26 15:15 /var/log/opscode/oc_bifrost/requests.log.1
-rw-r--r--. 1 opscode opscode 43390880 Mar 26 15:15 /var/log/opscode/opscode-erchef/requests.log.1
-rw-r--r--. 1 opscode opscode 31353 Mar 26 15:05 /var/log/opscode/redis_lb/current
-rw-r--r--. 1 opscode opscode 61612890 Mar 26 15:15 /var/log/opscode/nginx/access.log
************* Post-Reindex Log Dump *************
tail -10 /var/log/opscode/opscode-erchef/current
2021-03-22_12:30:19.12340 [info] Creating HTTP pool for Search Index.
2021-03-22_12:30:19.12869 [info] chef_keygen_cache configured size: 10 start_size: 0 max_workers: 4
2021-03-22_12:30:19.12920 [info] chef_keygen_cache starting with 0 keys in cache
2021-03-22_12:30:19.13288 [info] Application oc_chef_wm started on node 'erchef@127.0.0.1'
2021-03-22_12:30:19.13303 [info] Application oc_erchef started on node 'erchef@127.0.0.1'
2021-03-22_12:30:19.14346 [info] Application mnesia started on node 'erchef@127.0.0.1'
2021-03-22_12:30:19.14372 [info] Application eper started on node 'erchef@127.0.0.1'
2021-03-22_12:30:19.14391 [info] Application recon started on node 'erchef@127.0.0.1'
2021-03-22_12:30:19.14400 [info] Application observer_cli started on node 'erchef@127.0.0.1'
2021-03-22_12:30:19.14409 [info] Application efast_xs started on node 'erchef@127.0.0.1'
tail -3 /var/log/opscode/opscode-erchef/requests.log.1
2021-03-26T19:08:22Z erchef@127.0.0.1 method=GET; path=/organizations/abc/cookbook_artifacts/logrotate/ab4495244aeb1dab2fdd6fe883d3d53d55ef03cf; status=200; req_id=g3IAA2QAEGVyY2hlZkAxMjcuMC4wLjEBAAGzFj3UAAH2gp2U; org_name=abc; couchdb_groups=false; couchdb_organizations=false; couchdb_containers=false; couchdb_acls=false; couchdb_associations=false; couchdb_association_requests=false; req_time=11; rdbms_time=0; rdbms_count=3; authz_time=4; authz_count=1; user=myhost0020.abc.com; req_api_version=2; 
2021-03-26T19:08:25Z erchef@127.0.0.1 method=PUT; path=/organizations/abc/nodes/myhost0020.abc.com; status=200; req_id=g3IAA2QAEGVyY2hlZkAxMjcuMC4wLjEBAAGzrD3UAAH2gp2U; org_name=abc; couchdb_groups=false; couchdb_organizations=false; couchdb_containers=false; couchdb_acls=false; couchdb_associations=false; couchdb_association_requests=false; req_time=76; rdbms_time=9; rdbms_count=4; solr_time=31; solr_count=1; authz_time=4; authz_count=1; data_collector_time=8; data_collector_count=1; user=myhost0020.abc.com; req_api_version=2; 
2021-03-26T19:15:06Z erchef@127.0.0.1 method=GET; path=/organizations; status=200; req_id=g3IAA2QAEGVyY2hlZkAxMjcuMC4wLjEBAAHKpz3UAAH2gp2U; couchdb_groups=false; couchdb_organizations=false; couchdb_containers=false; couchdb_acls=false; couchdb_associations=false; couchdb_association_requests=false; req_time=9; rdbms_time=1; rdbms_count=3; authz_time=4; authz_count=1; user=pivotal; req_api_version=1; 
tail -10 /var/log/opscode/elasticsearch/current
2021-03-24_19:35:03.91252 [2021-03-24T15:35:03,912][INFO ][o.e.t.TransportService   ] [myhost0010.abc.com] publish_address {127.0.0.1:9300}, bound_addresses {127.0.0.1:9300}
2021-03-24_19:35:06.98563 [2021-03-24T15:35:06,984][INFO ][o.e.c.s.MasterService    ] [myhost0010.abc.com] zen-disco-elected-as-master ([0] nodes joined), reason: new_master {myhost0010.abc.com}{SY86b8oBTLeJBpJC5XWFmw}{ymTE7rjjQBiAQKfb08agpQ}{127.0.0.1}{127.0.0.1:9300}
2021-03-24_19:35:06.99124 [2021-03-24T15:35:06,990][INFO ][o.e.c.s.ClusterApplierService] [myhost0010.abc.com] new_master {myhost0010.abc.com}{SY86b8oBTLeJBpJC5XWFmw}{ymTE7rjjQBiAQKfb08agpQ}{127.0.0.1}{127.0.0.1:9300}, reason: apply cluster state (from master [master {myhost0010.abc.com}{SY86b8oBTLeJBpJC5XWFmw}{ymTE7rjjQBiAQKfb08agpQ}{127.0.0.1}{127.0.0.1:9300} committed version [1] source [zen-disco-elected-as-master ([0] nodes joined)]])
2021-03-24_19:35:07.00626 [2021-03-24T15:35:07,006][INFO ][o.e.h.n.Netty4HttpServerTransport] [myhost0010.abc.com] publish_address {127.0.0.1:9200}, bound_addresses {127.0.0.1:9200}
2021-03-24_19:35:07.00643 [2021-03-24T15:35:07,006][INFO ][o.e.n.Node               ] [myhost0010.abc.com] started
2021-03-24_19:35:07.10041 [2021-03-24T15:35:07,099][WARN ][o.e.d.i.m.AllFieldMapper ] [myhost0010.abc.com] [_all] is deprecated in 6.0+ and will be removed in 7.0. As a replacement, you can use [copy_to] on mapping fields to create your own catch all field.
2021-03-24_19:35:07.20312 [2021-03-24T15:35:07,202][INFO ][o.e.g.GatewayService     ] [myhost0010.abc.com] recovered [1] indices into cluster_state
2021-03-24_19:35:07.23083 [2021-03-24T15:35:07,230][WARN ][o.e.d.i.m.AllFieldMapper ] [myhost0010.abc.com] [_all] is deprecated in 6.0+ and will be removed in 7.0. As a replacement, you can use [copy_to] on mapping fields to create your own catch all field.
2021-03-24_19:35:07.58186 [2021-03-24T15:35:07,581][INFO ][o.e.c.r.a.AllocationService] [myhost0010.abc.com] Cluster health status changed from [RED] to [YELLOW] (reason: [shards started [[chef][4]] ...]).
2021-03-24_21:05:41.73089 [2021-03-24T17:05:41,730][INFO ][o.e.m.j.JvmGcMonitorService] [myhost0010.abc.com] [gc][5436] overhead, spent [286ms] collecting in the last [1s]
tail -10 /var/log/opscode/oc_bifrost/requests.log.1
2021-03-26T18:38:32Z oc_bifrost@127.0.0.1 method=GET; path=/objects/cfcb48c6610f80927890dbf32af7e392/acl/update/actors/f2a7009462e8340d0f30166b7e5c1294; status=200; requestor_id=f2a7009462e8340d0f30166b7e5c1294; req_time=2; rdbms.bifrost_db.has_permission_time=0; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
2021-03-26T19:05:05Z oc_bifrost@127.0.0.1 method=GET; path=/containers/cca5b7c269ec6a7250747d03297d9281/acl/read/actors/99b3b80bc7ee942d3d1a6d6be7cb6b15; status=200; requestor_id=99b3b80bc7ee942d3d1a6d6be7cb6b15; req_time=2; rdbms.bifrost_db.has_permission_time=1; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
2021-03-26T19:06:51Z oc_bifrost@127.0.0.1 method=GET; path=/objects/cfcb48c6610f80927890dbf32af7e392/acl/read/actors/f2a7009462e8340d0f30166b7e5c1294; status=200; requestor_id=f2a7009462e8340d0f30166b7e5c1294; req_time=2; rdbms.bifrost_db.has_permission_time=0; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
2021-03-26T19:06:51Z oc_bifrost@127.0.0.1 method=GET; path=/objects/fcc9e7d5e404d8516badc42e24334f66/acl/read/actors/f2a7009462e8340d0f30166b7e5c1294; status=200; requestor_id=f2a7009462e8340d0f30166b7e5c1294; req_time=2; rdbms.bifrost_db.has_permission_time=1; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
2021-03-26T19:06:51Z oc_bifrost@127.0.0.1 method=GET; path=/objects/85c9f427ebff72e106d4c80647f1d7f5/acl/read/actors/f2a7009462e8340d0f30166b7e5c1294; status=200; requestor_id=f2a7009462e8340d0f30166b7e5c1294; req_time=2; rdbms.bifrost_db.has_permission_time=1; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
2021-03-26T19:08:22Z oc_bifrost@127.0.0.1 method=GET; path=/objects/f31dffa6c900e46d2320f768ec03a23e/acl/read/actors/f2a7009462e8340d0f30166b7e5c1294; status=200; requestor_id=f2a7009462e8340d0f30166b7e5c1294; req_time=3; rdbms.bifrost_db.has_permission_time=1; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
2021-03-26T19:08:22Z oc_bifrost@127.0.0.1 method=GET; path=/objects/49e70386fe1be13ecf06aeb33ae78089/acl/read/actors/f2a7009462e8340d0f30166b7e5c1294; status=200; requestor_id=f2a7009462e8340d0f30166b7e5c1294; req_time=3; rdbms.bifrost_db.has_permission_time=1; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
2021-03-26T19:08:22Z oc_bifrost@127.0.0.1 method=GET; path=/objects/0e920ac5e01a99e2e73127df1b05f2b5/acl/read/actors/f2a7009462e8340d0f30166b7e5c1294; status=200; requestor_id=f2a7009462e8340d0f30166b7e5c1294; req_time=2; rdbms.bifrost_db.has_permission_time=1; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
2021-03-26T19:08:25Z oc_bifrost@127.0.0.1 method=GET; path=/objects/cfcb48c6610f80927890dbf32af7e392/acl/update/actors/f2a7009462e8340d0f30166b7e5c1294; status=200; requestor_id=f2a7009462e8340d0f30166b7e5c1294; req_time=2; rdbms.bifrost_db.has_permission_time=1; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
2021-03-26T19:15:06Z oc_bifrost@127.0.0.1 method=GET; path=/containers/cca5b7c269ec6a7250747d03297d9281/acl/read/actors/99b3b80bc7ee942d3d1a6d6be7cb6b15; status=200; requestor_id=99b3b80bc7ee942d3d1a6d6be7cb6b15; req_time=2; rdbms.bifrost_db.has_permission_time=0; rdbms.bifrost_db.has_permission_count=1; rdbms.bifrost_db.exists_time=0; rdbms.bifrost_db.exists_count=1; 
tail -10 /var/log/opscode/nginx/access.log
192.168.1.26 - - [2021-03-26T19:07:15+00:00]  "POST /organizations/abc/data-collector HTTP/1.1" 502 "0.015" 163 "-" "Chef Client/16.5.64 (ruby-2.7.1-p83; ohai-16.5.4; x64-mingw32; +https://chef.io)" "192.168.1.25:443" "502" "0.002" "16.5.64" "algorithm=sha256;version=1.3;" "myhost0020.abc.com" "2021-03-26T19:06:51Z" "MYppTKCkPKymF4aIYY6aRcK027ymFt91LAOT7yxygtg=" 1505 "1cb48bb6-d3a2-449a-9ef6-50b32a2c6f65"
192.168.1.26 - - [2021-03-26T19:07:32+00:00]  "POST /organizations/abc/data-collector HTTP/1.1" 502 "0.028" 163 "-" "Chef Client/16.5.64 (ruby-2.7.1-p83; ohai-16.5.4; x64-mingw32; +https://chef.io)" "192.168.1.25:443" "502" "0.006" "16.5.64" "algorithm=sha256;version=1.3;" "myhost0020.abc.com" "2021-03-26T19:06:51Z" "MYppTKCkPKymF4aIYY6aRcK027ymFt91LAOT7yxygtg=" 1505 "1cb48bb6-d3a2-449a-9ef6-50b32a2c6f65"
192.168.1.26 - - [2021-03-26T19:08:22+00:00]  "POST /organizations/abc/data-collector HTTP/1.1" 502 "0.015" 163 "-" "Chef Client/16.5.64 (ruby-2.7.1-p83; ohai-16.5.4; x64-mingw32; +https://chef.io)" "192.168.1.25:443" "502" "0.001" "16.5.64" "algorithm=sha256;version=1.3;" "myhost0020.abc.com" "2021-03-26T19:06:51Z" "MYppTKCkPKymF4aIYY6aRcK027ymFt91LAOT7yxygtg=" 1505 "1cb48bb6-d3a2-449a-9ef6-50b32a2c6f65"
192.168.1.26 - - [2021-03-26T19:08:22+00:00]  "POST /organizations/abc/reports/nodes/myhost0020.abc.com/runs HTTP/1.1" 404 "0.015" 558 "-" "Chef Client/16.5.64 (ruby-2.7.1-p83; ohai-16.5.4; x64-mingw32; +https://chef.io)" "-" "-" "-" "16.5.64" "algorithm=sha256;version=1.3;" "myhost0020.abc.com" "2021-03-26T19:08:21Z" "1AyUj35mm2qcq7ordeyZ/jTRHm+GkSixRJID2rE30LQ=" 1212 "1cb48bb6-d3a2-449a-9ef6-50b32a2c6f65"
192.168.1.26 - - [2021-03-26T19:08:22+00:00]  "GET /organizations/abc/cookbook_artifacts/chef-client/b2437a13eef749774828d032c2f1ff34e57582cd HTTP/1.1" 200 "0.020" 3425 "-" "Chef Client/16.5.64 (ruby-2.7.1-p83; ohai-16.5.4; x64-mingw32; +https://chef.io)" "127.0.0.1:8000" "200" "0.020" "16.5.64" "algorithm=sha256;version=1.3;" "myhost0020.abc.com" "2021-03-26T19:08:21Z" "47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=" 1137 "1cb48bb6-d3a2-449a-9ef6-50b32a2c6f65"
192.168.1.26 - - [2021-03-26T19:08:22+00:00]  "GET /organizations/abc/cookbook_artifacts/cron/566d56169d66a7456e34ff4fa0b9e3e6ef49493d HTTP/1.1" 200 "0.015" 1390 "-" "Chef Client/16.5.64 (ruby-2.7.1-p83; ohai-16.5.4; x64-mingw32; +https://chef.io)" "127.0.0.1:8000" "200" "0.014" "16.5.64" "algorithm=sha256;version=1.3;" "myhost0020.abc.com" "2021-03-26T19:08:22Z" "47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=" 1130 "1cb48bb6-d3a2-449a-9ef6-50b32a2c6f65"
192.168.1.26 - - [2021-03-26T19:08:22+00:00]  "GET /organizations/abc/cookbook_artifacts/logrotate/ab4495244aeb1dab2fdd6fe883d3d53d55ef03cf HTTP/1.1" 200 "0.014" 1416 "-" "Chef Client/16.5.64 (ruby-2.7.1-p83; ohai-16.5.4; x64-mingw32; +https://chef.io)" "127.0.0.1:8000" "200" "0.013" "16.5.64" "algorithm=sha256;version=1.3;" "myhost0020.abc.com" "2021-03-26T19:08:22Z" "47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=" 1135 "1cb48bb6-d3a2-449a-9ef6-50b32a2c6f65"
192.168.1.26 - - [2021-03-26T19:08:24+00:00]  "GET /organizations/abc/required_recipe HTTP/1.1" 404 "0.000" 558 "-" "Chef Client/16.5.64 (ruby-2.7.1-p83; ohai-16.5.4; x64-mingw32; +https://chef.io)" "-" "-" "-" "16.5.64" "algorithm=sha256;version=1.3;" "myhost0020.abc.com" "2021-03-26T19:08:24Z" "47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=" 1081 "1cb48bb6-d3a2-449a-9ef6-50b32a2c6f65"
192.168.1.26 - - [2021-03-26T19:08:25+00:00]  "PUT /organizations/abc/nodes/myhost0020.abc.com HTTP/1.1" 200 "0.093" 10044 "-" "Chef Client/16.5.64 (ruby-2.7.1-p83; ohai-16.5.4; x64-mingw32; +https://chef.io)" "127.0.0.1:8000" "200" "0.080" "16.5.64" "algorithm=sha256;version=1.3;" "myhost0020.abc.com" "2021-03-26T19:08:24Z" "02BoT3emHYy+jLqlnitz6juJr1RDUAOUarVqlg1fmS4=" 32045 "1cb48bb6-d3a2-449a-9ef6-50b32a2c6f65"
127.0.0.1 - - [2021-03-26T19:15:06+00:00]  "GET /organizations HTTP/1.1" 200 "0.013" 72 "-" "Chef Client/15.15.1 (ruby-2.6.6-p146; ohai-15.12.0; x86_64-linux; +https://chef.io)" "127.0.0.1:8000" "200" "0.012" "15.15.1" "algorithm=sha256;version=1.3;" "pivotal" "2021-03-26T19:15:06Z" "47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=" 1030 "484976b6-1900-4739-bc21-6f09f5f2f0cb"
tail -10 /var/log/opscode/redis_lb/current
2021-03-26_18:41:14.05798 29097:M 26 Mar 2021 14:41:14.057 * 1 changes in 900 seconds. Saving...
2021-03-26_18:41:14.05824 29097:M 26 Mar 2021 14:41:14.058 * Background saving started by pid 17795
2021-03-26_18:41:14.07149 17795:C 26 Mar 2021 14:41:14.071 * DB saved on disk
2021-03-26_18:41:14.07192 17795:C 26 Mar 2021 14:41:14.071 * RDB: 0 MB of memory used by copy-on-write
2021-03-26_18:41:14.15862 29097:M 26 Mar 2021 14:41:14.158 * Background saving terminated with success
2021-03-26_19:05:05.56833 29097:M 26 Mar 2021 15:05:05.568 * 1 changes in 900 seconds. Saving...
2021-03-26_19:05:05.56876 29097:M 26 Mar 2021 15:05:05.568 * Background saving started by pid 19474
2021-03-26_19:05:05.57781 19474:C 26 Mar 2021 15:05:05.577 * DB saved on disk
2021-03-26_19:05:05.57821 19474:C 26 Mar 2021 15:05:05.578 * RDB: 0 MB of memory used by copy-on-write
2021-03-26_19:05:05.66968 29097:M 26 Mar 2021 15:05:05.669 * Background saving terminated with success
```

**Comment by vinay-satish** [comment]
Reply to the questions


Q. Is epmd running? Is the port for epmd open?

A. Yes, epmd is running:

# ps ax | grep epmd

4198 ? S 0:05 /opt/opscode/embedded/lib/erlang/erts-10.6/bin/epmd -daemon

# /opt/opscode/embedded/lib/erlang/erts-10.6/bin/epmd -names

epmd: up and running on port 4369 with data:

name erchef at port 43850

name bookshelf at port 46332

name oc_bifrost at port 39869

Q. Is localhost resolvable to 127.0.0.1 on their system?

A. Yes

Q. Is their machine on ipv4 or ipv6?

A. Both

# ip -6 addr

1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 state UNKNOWN qlen 1000

inet6 ::1/128 scope host

valid_lft forever preferred_lft forever

2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 state UP qlen 1000

inet6 fe80::21d:d8ff:fe01:93ee/64 scope link

valid_lft forever preferred_lft forever

# ip -4 addr

1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000

inet 127.0.0.1/8 scope host lo

valid_lft forever preferred_lft forever

2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000

inet 192.168.140.24/21 brd 192.168.147.255 scope global noprefixroute eth0

valid_lft forever preferred_lft forever

The Engineer made a statement: “Updating from 13 -> 14 implies they are moving away from rabbitmq. It would be good to confirm none of the rabbitmq processes are stuck / blocking epmd.”

Question: How do I do this?

**Comment by PrajaktaPurohit** [comment]
This will not fix the issue but give some guidance to the user: https://github.com/chef/chef-server/issues/2457

**Comment by PrajaktaPurohit** [comment]
https://github.com/chef/umbrella/pull/126

**Comment by PrajaktaPurohit** [comment]
Merged, the fixes will be available in release 14.5 of Chef Infra Server.

**Comment by PrajaktaPurohit** [comment]
https://github.com/chef/chef-server/pull/2557
